Mathlib/GroupTheory/Submonoid/Membership.lean

```diff
@@ -452,6 +452,14 @@ theorem powers_one : powers (1 : M) = ⊥ :=
   bot_unique <| powers_subset (one_mem _)
 #align submonoid.powers_one Submonoid.powers_one
 
+/-- The submonoid generated by an element is a group if that element has finite order. -/
+def groupOfPowers {x : M} {n : ℕ} (hpos : 0 < n) (hx : x ^ n = 1) : Group (powers x) where
+  inv x := x ^ (n - 1)
+  mul_left_inv y := Subtype.ext <| by
+    obtain ⟨_, k, rfl⟩ := y
+    dsimp
+    rw [← pow_succ', Nat.sub_add_cancel hpos, ← pow_mul, mul_comm, pow_mul, hx, one_pow]
+
```

Comment:
It would be cool to also show
```suggestion
  zpow z x := x ^ z.natMod n
  zpow_zero' z := by simp only [Int.natMod, Int.zero_emod, Int.toNat_zero, pow_zero]
  zpow_neg' n x := Subtype.ext <| by
    obtain ⟨_, k, rfl⟩ := x
    simp [←pow_mul]
    sorry
  zpow_succ' n x := Subtype.ext <| by
    obtain ⟨_, k, rfl⟩ := x
    simp [←pow_mul]
    sorry

```
which I'm pretty sure is true!

Reply:
Done!